contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -223,18 +223,18 @@ contract UniswapAnchoredView is UniswapConfig {
         address uniswapMarket = config.uniswapMarket;
         uint cumulativePrice = currentCumulativePrice(config);
 
-        Observation storage newObservation = newObservations[uniswapMarket];
-        Observation storage oldObservation = oldObservations[uniswapMarket];
+        Observation memory newObservation = newObservations[uniswapMarket];
+        Observation memory oldObservation = oldObservations[uniswapMarket];
 
-        // Update new and old observations if elapsed time is bigger or equal to anchor period
+        // Update new and old observations if elapsed time is greater than or equal to anchor period
         uint timeElapsed = block.timestamp - newObservation.timestamp;
         if (timeElapsed >= anchorPeriod) {
             emit UniswapWindowUpdate(uniswapMarket, oldObservation.timestamp, newObservation.timestamp, oldObservation.acc, newObservation.acc);
-            oldObservation.timestamp = newObservation.timestamp;
-            oldObservation.acc = newObservation.acc;
+            oldObservations[uniswapMarket].timestamp = newObservation.timestamp;
+            oldObservations[uniswapMarket].acc = newObservation.acc;
 
-            newObservation.timestamp = block.timestamp;
-            newObservation.acc = cumulativePrice;
+            newObservations[uniswapMarket].timestamp = block.timestamp;
+            newObservations[uniswapMarket].acc = cumulativePrice;
         }
         return (cumulativePrice, oldObservation.acc, oldObservation.timestamp);
```

Comment:
won't this change the return value here in cases where the window markers are moved? before it would have returned the updated value that is eventually in storage under `oldObservations[uniswapMarket]` , now it will return the value that was in oldObservation at the beginning of the function call

Reply:
Yeah this used to return the 'current' old one, now it returns the old one whenever its updated